server/src/main/java/keywhiz/ServiceModule.java

```diff
@@ -210,7 +218,14 @@ public ServiceModule(KeywhizConfig config, Environment environment) {
     return dbi.onDemand(AclDAO.class);
   }
 
-  @Provides @Singleton Authenticator<BasicCredentials, User> authenticator(KeywhizConfig config, DBI dbi) {
-    return config.getUserAuthenticatorFactory().build(dbi);
+  @Provides @Singleton DSLContext jooqContext(ManagedDataSource dataSource) throws SQLException {
+    return DSL.using(dataSource.getConnection());
+  }
+
+  // Should this be readonly jooqContext?
```

Comment:
Yeah readonly should be fine.  UserDAO originally had some insert functionality but that has been removed.